packages/core/src/actions.ts

```diff
@@ -69,6 +69,26 @@ export function resolveActionObject(
         return [state, a];
       }
     );
+  } else if (isArray(dereferencedAction)) {
+    const params = actionObject.params ?? {};
+    return createDynamicAction(
+      { type: ActionTypes.ActionGroup, params },
+      (_event, { state }) => {
+        const action: BaseActionObject = {
+          type: actionObject.type,
+          params: {
+            actions: toActionObjects(dereferencedAction).map((object) => ({
+              ...object,
+              params: {
+                ...object.params,
+                ...params
+              }
```

Comment:
Is this fine? It would allow outer `params` to override inner `params`:

```ts
const machine = createMachine(
  {
    entry: {
      type: "sayHello",
      params: { location: "entry" }
    },
    initial: "state1",
    states: {
      state1: {
        always: {
          target: "state2",
          actions: { type: "sayHello", params: { location: "state1" } }
        }
      },
      state2: {
        entry: "sayHello"
      }
    }
  },
  {
    actions: {
      sayHello: [{ type: "greet", params: { location: "somewhere" } }],
      greet: log(({ action }) => `Hello from ${action.params!.location}!`)
    }
  }
);

interpret(machine).start()

// Hello from entry!
// Hello from state1!
// Hello from somewhere!
```

Doubt this will be a common scenario but does this feel right?

Reply:
This makes sense to me 👍